### ISSUE TITLE
TH generates ambiguous smart constructors for phantom parameters

### ISSUE THREAD

**Issue by isovector** [isovector/polysemy#48]
The TH currently spits out smart constructors like this:

```haskell
data Foo e m a where
  FooC :: Foo e m ()

fooC :: forall e r. Member (Foo e) r => Sem r ()
fooC = send FooC
```

Uh oh! The usage of `FooC` here is ambiguous! The plugin will correctly determine what it should be (`send $ FooC @e`), but it'd be nicer if the TH just did the right thing to begin with.

We want to generate this:

```haskell
fooC :: forall e r. Member (Foo e) r => Sem r ()
fooC = send (FooC @e)
```

**Comment by googleson78** [comment]
```haskell
fooC :: forall e r. Member (Foo e) r => Sem r ()
fooC = send (FooC @e)
```
is also ambiguous? (according to `ghc`)

**Comment by isovector** [comment]
It is, but differently ambiguous! The code as its written today is ambiguous in the effect its trying to send, but the correct code is only ambiguous in terms of `AllowAmbiguousTypes` :)

**Comment by isovector** [comment]
Fixed by #67
